<h3>Offline HCV-GLUE</h3>
<p>
The HCV-GLUE resource can be used "offline" to organise and analyse sequence data on a private computer.  
Offline HCV-GLUE takes the form of a GLUE project build (a linked dataset and set of analysis functions).
This project build can be loaded into an instance of the GLUE engine. 
</p>
<p>
A certain level of Unix command-line computing experience is required to install and use Offline HCV-GLUE. 
Please follow the instructions below to set up and use offline HCV-GLUE on your computer. A working installation
of offline HCV-GLUE is capable of a range of analysis functions. Some examples are given below; the 
<a href="http://tools.glue.cvr.ac.uk">GLUE engine website</a> documents how other functions may be accessed. 

<ol>
	<li><a ng-click="scrollTo('installGlue')">Install the GLUE engine</a></li>
	<li><a ng-click="scrollTo('downloadBuild')">Download an HCV-GLUE project build</a></li>
	<li><a ng-click="scrollTo('installBuild')">Install the project build into the GLUE database</a></li>
	<li><a ng-click="scrollTo('checkIntegrations')">Check that the MAFFT, RAxML and BLAST+ integrations are working correctly</a></li>
	<li><a ng-click="scrollTo('ngsAnalysis')">Example use: Basic analysis of deep sequencing data</a></li>
	<li><a ng-click="scrollTo('combinedReport')">Example use: Combined genotyping and drug resistance analysis reports</a></li>
</ol>

<p>Please contact <a href="mailto:josh.singer@glasgow.ac.uk">Josh Singer</a> or post a question on the 
<a href="https://groups.google.com/forum/#!forum/glue-support">GLUE support forum</a> with any questions or 
queries about offline HCV-GLUE</p>

<h3 id="installGlue">Install the GLUE engine</h3>

Please follow the <a href="http://tools.glue.cvr.ac.uk/#/installation">GLUE installation instructions</a>. 
The BLAST+, MAFFT and RAxML integration steps are required for most uses of Offline HCV-GLUE.

<h3 id="downloadBuild">Download an HCV-GLUE project build</h3>

<p>
There are two alternative builds for Offline HCV-GLUE. The smaller build includes the minimum set of packages for 
genotyping and drug resistance analysis. The larger build also includes the NCBI-HCV-GLUE extension, i.e. HCV sequences 
derived from the NCBI nucleotide database and their metadata, the set of atypical substitutions and RAS frequency data.
</p>	
<table class="table table-striped table-bordered table-nonfluid">
	<thead>
		<colgroup>
	       <col span="1" style="width: 50%;">
	       <col span="1" style="width: 50%;">
	    </colgroup>
	</thead>
	<tbody>
		<tr>
			<td><b>HCV-GLUE core project + HCV-DRUG-RESISTANCE extension</b></td>
			<td>Download <a href="/hcv_glue_dbs/hcv_glue.sql.gz">hcv_glue.sql.gz</a></td>
		</tr>
		<tr>
			<td><b>HCV-GLUE core project + HCV-DRUG-RESISTANCE extension + NCBI-HCV-GLUE extension</b></td>
			<td>Download <a href="/hcv_glue_dbs/ncbi_hcv_glue.sql.gz">ncbi_hcv_glue.sql.gz</a></td>
		</tr>
	</tbody>
</table>

<h3 id="installBuild">Install the project build into the GLUE database</h3>
<p>
<b>Warning:</b> This step will overwrite any data currently in the GLUE database.
</p>
<p>
Unzip the package and load into your MySQL databse using a Unix command line, something like this:
<div hljs language="bash" no-escape>
$ gunzip -c ncbi_hcv_glue.sql.gz | /usr/local/mysql/bin/mysql --user=gluetools --password=glue12345 GLUE_TOOLS
</div>
</p>
<p>Start the GLUE command line and use the <code>list project</code> GLUE command to validate that the project installed.</p>
<p>
<div hljs language="bash" no-escape>
<pre>
% gluetools.sh
GLUE version 0.1.135
Mode path: /
GLUE> list project
+======+================================+
| name |          description           |
+======+================================+
| hcv  | Hepatitis C variation analysis |
+======+================================+
Projects found: 1</pre>
</div>
</p>

<h3 id="checkIntegrations">Check that the MAFFT, RAxML and BLAST+ integrations are working correctly</h3>
<p>
We will use the <code>maxLikelihoodGenotyper</code> module within HCV-GLUE to assign a genotype and
subtype for a sequence in a FASTA file. This will validate that MAFFT and RAxML are integrated and working correctly. 
</p>
<p>
Download the file <a ng-click="downloadResistanceGeno1()">resistanceGeno1.fasta</a> and make sure a copy
is present inside the directory from which you launched GLUE.</p>

<p>
Enter the GLUE commands indicated below and check the output.
<div hljs language="bash" no-escape>
<pre>
GLUE> project hcv 
OK
Mode path: /project/hcv
GLUE> module maxLikelihoodGenotyper genotype file --fileName resistanceGeno1.fasta
+===========+====================+===================+
| queryName | genotypeFinalClade | subtypeFinalClade |
+===========+====================+===================+
| EF407428  | AL_1               | AL_1a             |
+===========+====================+===================+</pre>
</div>
</p>
<p>
We will use the <code>fastaSequenceReporter</code> module within HCV-GLUE to translate the NS5A gene of this sequence to amino acids. 
This step uses BLAST+ to align the sequence with a reference, and so if the step works it validates that BLAST+ is integrated and working correctly.
 
<p>
Enter the GLUE command indicated below and check the output. Use 'Q' to exit the interactive table.
<div hljs language="bash" no-escape>
<pre>
GLUE> module fastaSequenceReporter amino-acid -i resistanceGeno1.fasta -r REF_MASTER_NC_004102 -f NS5A -t REF_1a_M62321
+============+=========+=========+===========+
| codonLabel | queryNt | acRefNt | aminoAcid |
+============+=========+=========+===========+
| 1          | 6164    | 6258    | S         |
| 2          | 6167    | 6261    | G         |
| 3          | 6170    | 6264    | S         |
| 4          | 6173    | 6267    | W         |
| 5          | 6176    | 6270    | L         |
| 6          | 6179    | 6273    | R         |
| 7          | 6182    | 6276    | D         |
| 8          | 6185    | 6279    | I         |
| 9          | 6188    | 6282    | W         |
| 10         | 6191    | 6285    | D         |
| 11         | 6194    | 6288    | W         |
| 12         | 6197    | 6291    | I         |
| 13         | 6200    | 6294    | C         |
| 14         | 6203    | 6297    | E         |
| 15         | 6206    | 6300    | V         |
+============+=========+=========+===========+
Rows 1 to 15 of 448 [F:first, L:last, P:prev, N:next, Q:quit]</pre>
</div>
</p>
<h3 id="ngsAnalysis">Example use: Basic analysis of deep sequencing data</h3>
<p>
Offline HCV-GLUE can be used for minority variant analysis of deep sequencing data in the form of SAM/BAM files 
We presume here that a suitable sequence assembly method has been used to generate the SAM/BAM file from the 
raw sequencing data.</p>

<p>
Download the file <a href="exampleSequences/ngsData.bam">ngsData.bam</a> and make sure a copy
is present inside the directory from which you launched GLUE.</p>

<p>We will use the <code>hcvDRSamReporter</code> module within HCV-GLUE to translate those 
reads within the file which map to the NS5A gene. The function will report the balance of 
different amino acids residues at different locations within this gene. 
</p>
<p>
Enter the GLUE command indicated below and check the output. Use 'Q' to exit the interactive table.
<div hljs language="bash" no-escape>
<pre>
GLUE> module hcvDRSamReporter amino-acid -i ngsData.bam -r REF_MASTER_NC_004102 -f NS5A -p
+============+==========+===============+===========+=============+============+
| codonLabel | samRefNt |ancConstrRefNt | aminoAcid | readsWithAA | pctAaReads |
+============+==========+===============+===========+=============+============+
| 1          | 6147     |6258           | Y         | 2           | 0.47       |
| 1          | 6147     |6258           | S         | 418         | 99.05      |
| 1          | 6147     |6258           | P         | 2           | 0.47       |
| 2          | 6150     |6261           | S         | 2           | 0.48       |
| 2          | 6150     |6261           | R         | 2           | 0.48       |
| 2          | 6150     |6261           | G         | 417         | 99.05      |
| 3          | 6153     |6264           | S         | 421         | 100.00     |
| 4          | 6156     |6267           | *         | 1           | 0.24       |
| 4          | 6156     |6267           | W         | 418         | 99.76      |
| 5          | 6159     |6270           | L         | 354         | 100.00     |
| 6          | 6162     |6273           | R         | 352         | 99.44      |
| 6          | 6162     |6273           | K         | 1           | 0.28       |
| 6          | 6162     |6273           | G         | 1           | 0.28       |
| 7          | 6165     |6276           | D         | 354         | 100.00     |
| 8          | 6168     |6279           | I         | 276         | 100.00     |
+============+==========+===============+===========+=============+============+
Rows 1 to 15 of 764 [F:first, L:last, P:prev, N:next, Q:quit]</pre></div>

<h3 id="combinedReport">Example use: Combined genotyping and drug resistance analysis reports</h3>

<p>Offline HCV-GLUE contains modules which run combined genotyping and drug resistance analysis procedures, 
and generate detailed reports in the form of XML or JSON documents. Like any GLUE function, these procedures 
can be run from the Unix command line and so integrated into a bioinformatics pipeline. </p>

This first example will produce a report for a FASTA file, e.g. <a ng-click="downloadResistanceGeno1()">resistanceGeno1.fasta</a>. 
A preview of the JSON output is available <a href="#/aboutGlueProject/exampleFastaReport">here</a>.
<div hljs language="bash" no-escape>
<pre>% gluetools.sh -p cmd-result-format:json -EC -i project hcv module hcvDRReporter invoke-function reportFasta resistanceGeno1.fasta</pre>
</div>
This example will produce a report for a SAM/BAM file, e.g. <a href="exampleSequences/ngsData.bam">ngsData.bam</a>. 
A preview of the JSON output is available <a href="#/aboutGlueProject/exampleBamReport">here</a>.
<div hljs language="bash" no-escape>
<pre>% gluetools.sh -p cmd-result-format:json -EC -i project hcv module hcvDRReporter invoke-function reportBam ngsData.bam</pre>
</div>

If XML output is required, use <code>cmd-result-format:xml</code>.

 