<h2>Offline HCV-GLUE</h2>
<p>
The HCV-GLUE resource can be used "offline" to organise and analyse sequence data on a private computer.  
Offline HCV-GLUE takes the form of a GLUE project build (a linked dataset and set of analysis functions).
This project build can be loaded into an instance of the GLUE engine. 
</p>
<p>
A certain level of Unix command-line computing experience is required to install and use Offline HCV-GLUE. 
Please follow the instructions below to set up and use offline HCV-GLUE on your computer. A working installation
of offline HCV-GLUE is capable of a range of analysis functions. Some examples are given below; the 
<a href="http://tools.glue.cvr.ac.uk">GLUE engine website</a> documents how other functions may be accessed. 

<ol>
	<li><a ng-click="scrollTo('installGlue')">Install the GLUE engine</a></li>
	<li><a ng-click="scrollTo('downloadInstallBuild')">Download and install the HCV-GLUE project build</a></li>
	<li><a ng-click="scrollTo('checkBuild')">Check the project build is working</a></li>
	<li><a ng-click="scrollTo('ngsAnalysis')">Example use: Basic analysis of deep sequencing data</a></li>
	<li><a ng-click="scrollTo('combinedReport')">Example use: Combined genotyping and drug resistance analysis reports</a></li>
</ol>

<p>Please contact <a href="mailto:josh.singer@glasgow.ac.uk">Josh Singer</a> or post a question on the 
<a href="https://groups.google.com/forum/#!forum/glue-support">GLUE support forum</a> with any questions or 
queries about offline HCV-GLUE</p>

<h3 id="installGlue">Install the GLUE engine</h3>

Please follow the <a href="http://tools.glue.cvr.ac.uk/#/installation">GLUE installation instructions</a>. 
We would strongly recommend a Docker-based installation of GLUE. 

<h3 id="downloadInstallBuild">Download and install the HCV-GLUE project build</h3>
Note: This step will erase anything that was previously in your database.

<h5><b>Docker-based GLUE</b></h5>
<p>
<ol>
<li>Make sure the <code>gluetools-mysql</code> container is running. </li>
<li>The following UNIX command will install the HCV-GLUE project build.
<div hljs language="js" no-escape>
$ docker exec gluetools-mysql installGlueProject.sh ncbi_hcv_glue
</div>
</li>
</ol>
</p>

<h5><b>Native GLUE</b></h5>
<p>
<ol>
<li>Download <a href="/hcv_glue_dbs/ncbi_hcv_glue.sql.gz">ncbi_hcv_glue.sql.gz</a></li>
<li>
Load the build into your MySQL databse using a Unix command line, adjusting the details according to your system:
<div hljs language="bash" no-escape>
$ gunzip -c ncbi_hcv_glue.sql.gz | /usr/local/mysql/bin/mysql --user=gluetools --password=glue12345 GLUE_TOOLS
</div>
</li>
</ol>
</p>

<h3 id="checkBuild">Check the project build is working</h3>
<p>
<ol>
<li>
Download the files <a ng-click="downloadResistanceGeno1()">resistanceGeno1.fasta</a> and 
<a href="exampleSequences/ngsData.bam">ngsData.bam</a> to a local directory, e.g. <code>/home/fred/glue_data</code>
</li>
<li>
Start the GLUE command line, ensuring that your local directory is accessible to GLUE:
<h5><b>Docker-based GLUE</b></h5>
<div hljs language="bash" no-escape>
$ docker run --rm -it --name gluetools -v /home/fred/glue_data:/home/fred/glue_data -w /home/fred/glue_data --link gluetools-mysql cvrbioinformatics/gluetools:latest
</div>
<h5><b>Native GLUE</b></h5>
<div hljs language="bash" no-escape>
$ cd /home/fred/glue_data
$ gluetools.sh
</div>
</li>
<li>
Use <code>list project</code> to check that the HCV project is there.
<p>
<pre>
Mode path: /
GLUE> list project
+======+================================+
| name |          description           |
+======+================================+
| hcv  | Hepatitis C variation analysis |
+======+================================+
Projects found: 1</pre>
</div>
</li>
</p>
<li>
<p>
We will use the <code>maxLikelihoodGenotyper</code> module within HCV-GLUE to assign a genotype and
subtype for the sequence in the FASTA file. Enter the GLUE commands below and check the output.
<div hljs language="bash" no-escape>
<pre>
GLUE> project hcv 
OK
Mode path: /project/hcv
GLUE> module maxLikelihoodGenotyper genotype file --fileName resistanceGeno1.fasta
+===========+====================+===================+
| queryName | genotypeFinalClade | subtypeFinalClade |
+===========+====================+===================+
| EF407428  | AL_1               | AL_1a             |
+===========+====================+===================+</pre>
</div>
</p>
</li>
<li>
<p>
We will use the <code>phdrFastaSequenceReporter</code> module within HCV-GLUE to translate the NS5A gene of this sequence to amino acids. 
Enter the GLUE command below and check the output. Use 'Q' to exit the interactive table.
<div hljs language="bash" no-escape>
<pre>
Mode path: /project/hcv
GLUE> module phdrFastaSequenceReporter amino-acid -i resistanceGeno1.fasta -r REF_MASTER_NC_004102 -f NS5A -t REF_1a_M62321 -a AL_UNCONSTRAINED
+==========+=========+==========+==========+===========+===========+===========+
|codonLabel| queryNt | relRefNt | codonNts | aminoAcid |definiteAas|possibleAas|
+==========+=========+==========+==========+===========+===========+===========+
|1         | 6164    | 6258     | TCC      | S         |S          |S          |
|2         | 6167    | 6261     | GGC      | G         |G          |G          |
|3         | 6170    | 6264     | TCC      | S         |S          |S          |
|4         | 6173    | 6267     | TGG      | W         |W          |W          |
|5         | 6176    | 6270     | CTA      | L         |L          |L          |
|6         | 6179    | 6273     | AGG      | R         |R          |R          |
|7         | 6182    | 6276     | GAC      | D         |D          |D          |
|8         | 6185    | 6279     | ATC      | I         |I          |I          |
|9         | 6188    | 6282     | TGG      | W         |W          |W          |
|10        | 6191    | 6285     | GAC      | D         |D          |D          |
|11        | 6194    | 6288     | TGG      | W         |W          |W          |
|12        | 6197    | 6291     | ATA      | I         |I          |I          |
|13        | 6200    | 6294     | TGC      | C         |C          |C          |
|14        | 6203    | 6297     | GAG      | E         |E          |E          |
|15        | 6206    | 6300     | GTG      | V         |V          |V          |
|16        | 6209    | 6303     | TTG      | L         |L          |L          |
|17        | 6212    | 6306     | AGC      | S         |S          |S          |
|18        | 6215    | 6309     | GAC      | D         |D          |D          |
|19        | 6218    | 6312     | TTT      | F         |F          |F          |
+==========+=========+==========+==========+===========+===========+===========+
Rows 1 to 19 of 448 [F:first, L:last, P:prev, N:next, Q:quit]</pre>
</div>
</p>
</li>
</ol>
<h3 id="ngsAnalysis">Example use: Basic analysis of deep sequencing data</h3>
<p>
Offline HCV-GLUE can be used for minority variant analysis of deep sequencing data in the form of SAM/BAM files 
We presume here that a suitable sequence assembly method has been used to generate the SAM/BAM file from the 
raw sequencing data.</p>

<p>We will use the <code>phdrSamReporter</code> module within HCV-GLUE to translate those 
reads within the file which map to the NS5A gene. The function will report the balance of 
different amino acids residues at different locations within this gene. 
</p>
<p>
Enter the GLUE command below and check the output. Use 'Q' to exit the interactive table.
<div hljs language="bash" no-escape>
<pre>
GLUE> module phdrSamReporter amino-acid -i ngsData.bam -r REF_MASTER_NC_004102 -f NS5A -p -a AL_UNCONSTRAINED
+============+==========+==========+===========+=============+============+
| codonLabel | samRefNt | relRefNt | aminoAcid | readsWithAA | pctAaReads |
+============+==========+==========+===========+=============+============+
| 1          | 6147     | 6258     | Y         | 1           | 0.32       |
| 1          | 6147     | 6258     | S         | 314         | 99.37      |
| 1          | 6147     | 6258     | P         | 1           | 0.32       |
| 2          | 6150     | 6261     | S         | 1           | 0.32       |
| 2          | 6150     | 6261     | R         | 1           | 0.32       |
| 2          | 6150     | 6261     | G         | 313         | 99.37      |
| 3          | 6153     | 6264     | S         | 315         | 100.00     |
| 4          | 6156     | 6267     | W         | 315         | 100.00     |
| 5          | 6159     | 6270     | L         | 314         | 100.00     |
| 6          | 6162     | 6273     | R         | 312         | 99.36      |
| 6          | 6162     | 6273     | K         | 1           | 0.32       |
| 6          | 6162     | 6273     | G         | 1           | 0.32       |
| 7          | 6165     | 6276     | D         | 314         | 100.00     |
| 8          | 6168     | 6279     | I         | 274         | 100.00     |
| 9          | 6171     | 6282     | W         | 274         | 100.00     |
| 10         | 6174     | 6285     | D         | 253         | 100.00     |
| 11         | 6177     | 6288     | *         | 1           | 0.40       |
| 11         | 6177     | 6288     | W         | 252         | 99.60      |
| 12         | 6180     | 6291     | I         | 197         | 100.00     |
+============+==========+==========+===========+=============+============+
Rows 1 to 19 of 753 [F:first, L:last, P:prev, N:next, Q:quit]</pre></div>

<h3 id="combinedReport">Example use: Combined genotyping and drug resistance analysis reports</h3>

<p>Offline HCV-GLUE contains modules which run combined genotyping and drug resistance analysis procedures, 
and generate detailed reports.<p>

<p>
This first example will read in the FASTA file, <code>resistanceGeno1.fasta</code> and produce an HTML report file <code>resistanceGeno1.html</code>. 
Click <a href="exampleReports/resistanceGeno1.html?hcvGlueVersion=0.1.54" target="_blank">here</a> for a preview of the report. 
<div hljs language="bash" no-escape>
<pre>
Mode path: /project/hcv
GLUE> module phdrReportingController invoke-function reportFastaAsHtml resistanceGeno1.fasta resistanceGeno1.html
</pre>
</div>
</p>

<p>
The second example will read in the BAM file, <code>ngsData.bam</code> and produce a new HTML report <code>ngsData.html</code>, 
using a 15% reads percentage minimum threshold for reporting polymorphisms.
Click <a href="exampleReports/ngsData.html?hcvGlueVersion=0.1.54" target="_blank">here</a> for a preview of the report. 

<div hljs language="bash" no-escape>
<pre>
Mode path: /project/hcv
GLUE> module phdrReportingController invoke-function reportBamAsHtml ngsData.bam 15.0 ngsData.html
</pre>
</div>

<p>HCV-GLUE can also generate files containing genotyping and drug resistance reporting data in a machine-readable format (XML or JSON). 
These could be used e.g. for integration into a bioinformatics pipeline. Contact the team if this is of interest.</p>

 